src/audio/kpb.c

```diff
@@ -897,6 +897,124 @@ static int kpb_reset(struct comp_dev *dev)
 	return ret;
 }
 
+#ifdef KPB_HIFI3
+static void kpb_micselect_copy(struct comp_dev *dev, struct comp_buffer __sparse_cache *sink,
```

Comment:
It would be nicer to split the HiFiX function versions to separate files like kpb_hifiX.c and kpb_generic.c. The processing modes become too cluttered if with all the versions under #if macros.